controllers/kserve_ray_tls_controller.go

```diff
@@ -0,0 +1,412 @@
+			return err
+		}
+	}
+
+	if shouldDeletedRaySecret(existingSecretResource, noMultiNodeSrExistInNs) {
+		hasChanged = true
+		log.V(1).Info("Delta found", "remove", existingSecretResource.GetName(), "namespace", existingSecretResource.Namespace)
+		if err = r.client.Delete(ctx, existingSecretResource); err != nil {
+			return err
+		}
+	}
+	if !hasChanged && !noMultiNodeSrExistInNs {
+		log.V(1).Info("No delta found", "name", desiredSecretResource.GetName(), "namespace", desiredSecretResource.Namespace)
+	}
+	return nil
+}
+
+func shouldAddRaySecret(existingSecret *corev1.Secret, noMultiNodeSrExistInNs bool) bool {
+	return !noMultiNodeSrExistInNs && utils.IsNil(existingSecret)
+}
+func isUpdatedRaySecret(desiredSecret *corev1.Secret, existingSecret *corev1.Secret) bool {
+	return utils.IsNotNil(existingSecret) && !reflect.DeepEqual(desiredSecret.Data, existingSecret.Data)
+}
+func shouldDeletedRaySecret(existingSecret *corev1.Secret, noMultiNodeSrExistInNs bool) bool {
+	return utils.IsNotNil(existingSecret) && noMultiNodeSrExistInNs
+}
+
+func (r *KServeRayTlsReconciler) cleanupRayResourcesByKind(ctx context.Context, log logr.Logger, targetNs string, kind string) error {
+	if kind == "ConfigMap" {
+		configmap := &corev1.ConfigMap{}
+		err := r.client.Get(ctx, types.NamespacedName{
+			Name:      constants.RayTlsScriptConfigMapName,
+			Namespace: targetNs,
+		}, configmap)
+		if err != nil {
+			if apierrs.IsNotFound(err) {
+				log.Info("ConfigMap not found, skipping", "name", constants.RayTlsScriptConfigMapName, "namespace", targetNs)
+			}
+			return err
+		}
+
+		log.Info("Deleting ConfigMap", "name", constants.RayTlsScriptConfigMapName, "namespace", targetNs)
+		err = r.client.Delete(ctx, configmap)
+		if err != nil {
+			return err
+		}
+	}
+
+	if kind == "Secret" {
+		secret := &corev1.Secret{}
+		err := r.client.Get(ctx, types.NamespacedName{
+			Name:      constants.RayCASecretName,
+			Namespace: targetNs,
+		}, secret)
+		if err != nil {
+			if apierrs.IsNotFound(err) {
+				log.Info("Secret not found, skipping", "name", constants.RayCASecretName, "namespace", targetNs)
+			}
+			return err
+		}
+
+		log.Info("Deleting Secret", "name", constants.RayCASecretName, "namespace", targetNs)
+		err = r.client.Delete(ctx, secret)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func existMultiNodeServingRuntimeInNs(targetNs string, srList kservev1alpha1.ServingRuntimeList) bool {
+	for _, sr := range srList.Items {
+		if sr.Namespace == targetNs {
+			return isMultiNodeServingRuntime(sr)
+		}
+	}
+	return false
+}
+
+// Determine if ServingRuntime matches specific conditions
+// TO-DO upstream Kserve 0.15 will have a new API WorkerSpec
+// So for now, it will check servingRuntime name, but after we move to 0.15, it needs to check workerSpec is specified or not.(RHOAIENG-16147)
+func isMultiNodeServingRuntime(servingRuntime kservev1alpha1.ServingRuntime) bool {
+	return servingRuntime.Name == "vllm-multinode-runtime"
```

Comment:
Perhaps, slightly better to check for the presence of the multi-node fields (was it the workerSpec field?), like in KServe to keep it generic.